service/history/replication/task_processor.go

```diff
@@ -117,24 +119,23 @@ func NewTaskProcessor(
 	replicationTaskExecutor TaskExecutor,
 	eventSerializer serialization.Serializer,
 ) TaskProcessor {
-	shardID := shard.GetShardID()
-	taskRetryPolicy := backoff.NewExponentialRetryPolicy(config.ReplicationTaskProcessorErrorRetryWait(shardID)).
-		WithBackoffCoefficient(config.ReplicationTaskProcessorErrorRetryBackoffCoefficient(shardID)).
-		WithMaximumInterval(config.ReplicationTaskProcessorErrorRetryMaxInterval(shardID)).
-		WithMaximumAttempts(config.ReplicationTaskProcessorErrorRetryMaxAttempts(shardID)).
-		WithExpirationInterval(config.ReplicationTaskProcessorErrorRetryExpiration(shardID))
+	taskRetryPolicy := backoff.NewExponentialRetryPolicy(config.ReplicationTaskProcessorErrorRetryWait(pollingShardID)).
```

Comment:
This config cannot be based on pollingShardID, right? or why would it be?

Reply:
if you have 3 clusters and the shard counts are all different, then what will you specify in the config?

Reply:
Yes. You are right. I am going to revert this.